### PR TITLE
Event Parameters Package and String representations.

### PR DESCRIPTION
**Pull request by jessetilro**
The name is this branch is misleading: I was planning to integrate event parameters with the networking components, but too many small things required my attention before I could continue. I reorganized the event package, by distributing the components over different subpackages. Also I updated the EventReader to read the different parameters and their description for each Event type from the data file. I used this to create proper toString methods for EventParameterValues, EventParameter, EventParameterCollection and Event, in order to be able to easily present the Player with the right information.

Next PR will make sure parameters are synchronized over the network properly, I promise.

Contributes to resolving Issue #190.
Fixed some comment made on #213.

### CODE REVIEW

**Comment by jessetilro** [comment]
My sincere apologies for the failing builds. It is ironic, because CheckStyle urgently requested me to make the defective line shorter, eliminating a static error while introducing a dynamic one...

**Comment by dmalarev** [comment]
code looks good. Nicely done

File: src/test/java/nl/tudelft/pixelperfect/event/parameter/EventParameterTest.java [hunk 1]
```diff
@@ -152,4 +152,28 @@ public void testHashCodeSame() {
...
   }
 
+  /**
+   * The toString method should generate a correct String representation of a generic
+   * EventParameter.
+   */
+  @Test
+  public void testToStringGeneric() {
+    objectGeneric
+        .setSummary("The Answer to the Ultimate Question of Life, the Universe, and Everything");
```
Comment (dalderliesten) [inline src/test/java/nl/tudelft/pixelperfect/event/parameter/EventParameterTest.java]: Is such a specific test really needed?

**Comment by dalderliesten** [comment]
Looks fine to me.

File: src/test/java/nl/tudelft/pixelperfect/event/parameter/EventParameterTest.java (hunk 1 above)
Comment (jessetilro) [inline src/test/java/nl/tudelft/pixelperfect/event/parameter/EventParameterTest.java]: No, but it makes testing more fun, which is important given the fact that we need to spend a lot of time on this.

**Comment by FlorisXll** [comment]
LGTM

**Pull request merged**